### DISCUSSION TITLE
Trackball stops working

### DISCUSSION THREAD

**Discussion by minh-tech** [Yowkees/keyball#144] [Q&A] [answered]
Hi Yowkees,

Today I flashed default qmk firmware for keyball61 using QMK tool. Before flashing, the trackball still worked, but after that, it stopped working, I can still type on keyboard, so only trackball not working. I could use trackball before flashing so may not because of soldering
Can you please help me with this issue?

https://user-images.githubusercontent.com/44096812/194974463-c637e573-c3b5-4579-b5e2-1ee0c8315cc5.mp4

![JPEG image 3](https://user-images.githubusercontent.com/44096812/194975067-9028eb05-7153-4695-920b-c03496d540b3.jpeg)
![JPEG image 2](https://user-images.githubusercontent.com/44096812/194975069-6970d888-164b-43fb-a939-05046e221506.jpeg)
![JPEG image 4](https://user-images.githubusercontent.com/44096812/194975072-df694cf2-ad4e-41d6-b339-4e57235d7e93.jpeg)
![JPEG image](https://user-images.githubusercontent.com/44096812/194975078-0b1ce1cd-e501-4fca-b100-deaf35467558.jpeg)

**Comment by Yowkees** [comment]
Hi!

I saw your soldering. Looks good.

Could you try to flash again by using Remap? link is below.

https://remap-keys.app/catalog/RZSU1CrvEW4lns0ww5BM/firmware

**Comment by minh-tech** [comment]
Hi,
I just flashed all 3 firmware versions from the REMAP, but the trackball doesn't move at all.
I also re-soldered 4 jumpers near the trackball but nothing changes. I see voltage on some pins of PMW3360DM‐T2QU by using Digital Multimeter. How can I know if the PMW3360 alive?

**Comment by minh-tech** [comment]
I decided to use a new pro micro controller and flash it from the beginning. After flashing the first time, the mouse moved around crazily by itself. I unplug the USB cable and re-plug it, now it stopped working again. I believe there is something wrong with PMW3360 or the board. Do you have any suggestion?

**Reply by Yowkees** [reply]
In that case, we have to check sordering of optical IC. 
Can I see more big (close) picture of IC soldering?

**Comment by minh-tech** [comment]
Sure, here is the pictures
![JPEG image 5](https://user-images.githubusercontent.com/44096812/195097490-d6bb5d66-8151-49fc-ad1c-72b897d80e10.jpeg)
![JPEG image 7](https://user-images.githubusercontent.com/44096812/195097492-39f44808-27f9-4a02-9153-e8a142c9802b.jpeg)
![JPEG image 6](https://user-images.githubusercontent.com/44096812/195097496-baec3c36-311b-4ec1-810c-82d52d9314c5.jpeg)

**Comment by Yowkees** [comment, marked as answer]
Thank you for showing soldering. I think it's fine.
By checking picture No.3, soldering is good.

Oh, you have Digital multimeter. please check whether optical IC is working by my gerber data showing below.
https://github.com/Yowkees/keyball/tree/main/keyball46/ball_design_data

One more, I am afraided you are using purple ProMicro.
I have experiences to use 2 types of ProMicro below links.
https://shirogane-lab.com/products/promicro-type-b?variant=40385685848109
https://shirogane-lab.com/products/promicro-type-b?variant=40385685782573

I am not sure that purple ProMicro provides how much voltage or something.
It could be related to this problem.

Anyway, let me know how was your research. Thank you very much.

**Comment by minh-tech** [comment]
Thank you for your ball design file. I used Digital multimeter to check every pin of the IC and finally figured out the pin 8 had a bad soldering, it can not be seen with the eyes. I cleaned the old soldering and apply a new one, now the trackball is working again.
I really appreciate your help and quick response.

I always hope to have a keyboard integrated a trackball, finally I found it. Thank you for your hard work. 
Wish your store more and more development and release more keyboards.

**Comment by Yowkees** [comment]
Wow! 
I am very glad that the data is useful to you.

You have great electrical skil!!
Thank you very much for your hard working.

I'll do my best to continue selling Keyball in the future.
Thank you:-)
